scipy/interpolate/tests/test_interpolate.py

```diff
@@ -2613,6 +2613,28 @@ def test_broadcastable_input(self):
                          interp_points3.size ==
                          interp_points4.size, True)
 
+    def test_read_only(self):
+        # input data
+        np.random.seed(0)
+        xy = np.random.random((10, 2))
+        x, y = xy[:, 0], xy[:, 1]
+        z = np.hypot(x, y)
+        
+        # x-y grid for interpolation
+        X = np.linspace(min(x), max(x))
+        Y = np.linspace(min(y), max(y))
+        X, Y = np.meshgrid(X, Y)
+        XY = np.vstack((X.ravel(), Y.ravel())).T
```

Comment:
Now I think you're getting away with it because `XY` isn't contiguous. `xi` should need to be marked `const`.

Reply:
Sorry for many time. I didn't realize `XY` is not contiguous. I added `const` for `xi`. PTAL.